### PR TITLE
fix(python): raise error when memory:// in parquet path

### PR DESCRIPTION
**Pull request by deanm0000**
fixes https://github.com/pola-rs/polars/issues/14417

### CODE REVIEW

File: py-polars/polars/io/parquet/functions.py [hunk 1]
```diff
@@ -209,7 +211,8 @@ def read_parquet_schema(source: str | Path | IO[bytes] | bytes) -> dict[str, Dat
     """
     if isinstance(source, (str, Path)):
         source = normalize_filepath(source)
-
+    print(type(source))
+    print(dir(source))
```
Comment (stinodego) [inline py-polars/polars/io/parquet/functions.py:215]: These should be deleted.

**Review by stinodego — CHANGES_REQUESTED** [review]
Thanks for the PR, but I don't think this is the way to fix it. We don't support fsspec directly anymore in `read_parquet`. Rather than support this, it should error.

**Comment by deanm0000** [comment]
@stinodego changes made.

**Review by stinodego — COMMENTED** [review]
This fix seems much too ad hoc. Where does the infinite loop come from? We have to figure that out, rather than special case any prefixes that end up causing issues.

**Comment by deanm0000** [comment]
The loop is goes like this:

`read_parquet` will call `scan_parquet` if the source isn't `(io.BufferedIOBase, io.RawIOBase, bytes)`

`scan_parquet` calls `LazyFrame::scan_parquet` which then calls `polars.io.parquet.anonymous_scan._scan_parquet_fsspec` which calls `_scan_parquet_impl` which in turn calls `read_parquet`.

I don't know where you'd prefer to break the loop. I'd guess here:

https://github.com/pola-rs/polars/blob/286744a0a466070dfa2df628fb9059d802027a42/py-polars/polars/io/parquet/anonymous_scan.py#L18-L21


I'm also confused at your comment

>We don't support fsspec directly anymore in read_parquet

because there are a lot of references to fsspec and my original fix just piggy backed off an existing `if _FSSPEC_AVAILABLE` block

**Pull request closed without merging**